scripts/extensions/pool_update.apply

```diff
@@ -118,7 +117,11 @@ if __name__ == '__main__':
             print(failure_message(PATCH_APPLY_FAILED,
                                   ['Apply failed: %s' % e]))
         finally:
-            session.xenapi.pool_update.detach(update)
+            try:
+                session.xenapi.pool_update.detach(update)
+            except Exception as e:
+                print(failure_message(PATCH_APPLY_FAILED,
+                    ['Apply failed: %s' % e]))
```

Comment:
I approve of "Failed to apply the update."
How does the message look by the time the user sees it though? The argument we are giving to the `print` function is a failure_message which already has PATCH_APPLY_FAILED in it, and this should be translated into an appropriate human-readable string anyway. The additional string we are writing here is for additional detail, so can we just use `'%s' e` (i.e. string of `e`) on its own for this string? I suppose the one detail added by the text suggested earlier is that we were trying to apply an update rather than a patch, but I'm not sure that's particularly valuable.